### PR TITLE
interfaces/builtin: use permanent snippets in bool-file

### PR DESCRIPTION
**Pull request by zyga**
This patch changes the security of bool-file to align it with the
permanent vs connected snippet idea. The slot is now permanently able to
control GPIOs (to export/unexport them and to change direction).

The plug side remains the same, that is, plugs can access the GPIO value
or LED brightness, as before.

This also has a nice property of not requiring any restarts of the
service that has this slot when a plug is disconnected because the
plug-side security is not affected by connections. This doesn't matter
yet but will quite soon as interface code is being integrated with the
overlord.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

**Comment by zyga** [comment]
retest this please

File: interfaces/builtin/bool_file.go [hunk 1]
```diff
@@ -77,10 +77,20 @@ func (iface *BoolFileInterface) SanitizePlug(slot *interfaces.Plug) error {
 	return nil
 }
 
-// ConnectedSlotSnippet returns the configuration snippet required to provide a bool-file interface.
-// Producers gain control over exporting, importing GPIOs as well as
-// controlling the direction of particular pins.
+// ConnectedSlotSnippet returns security snippet specific to a given connection between the bool-file slot and some plug.
+// Applications associated with the slot don't gain any extra permissions.
```
Comment (chipaca) [inline interfaces/builtin/bool_file.go]: Should this still talk about `bool-file`?

Comment (chipaca) [inline interfaces/builtin/bool_file.go]: d'oh, it's a method of it.

Should the method still exist?

**Comment by chipaca** [comment]
looks reasonable to me. I'm assuming the `...Snippet` methods are part of an interface.

File: interfaces/builtin/bool_file.go (hunk 1 above)
Comment (zyga) [inline interfaces/builtin/bool_file.go]: Yes, we have four snippet methods {connected,permanent}x{slot,plug}

**Comment by niemeyer** [comment]
LGTM

**Pull request merged**